### PR TITLE
Fix/9.0/account/fiscalyear sequence

### PR DESCRIPTION
**Pull request by StefanRijnhart**

### CODE REVIEW

File: addons/account/migrations/9.0.1.1/post-migration.py [hunk 1]
```diff
@@ -336,6 +336,31 @@ def map_account_tax_template_type(cr):
         table='account_tax_template', write='sql')
 
 
+def migrate_account_sequence_fiscalyear(cr):
+    """ Migrate subsequences from fiscalyears to sequence date ranges
+    so that the next number is aligned with the last assigned number """
+    openupgrade.logged_query(
+        cr,
+        """ \
```
Comment (pedrobaeza) [inline addons/account/migrations/9.0.1.1/post-migration.py:344]: Why the `\`?

File: addons/account/migrations/9.0.1.1/post-migration.py [hunk 2]
```diff
@@ -336,6 +336,31 @@ def map_account_tax_template_type(cr):
...
+        WHERE asf.fiscalyear_id = af.id
+             and asf.sequence_id = irs.id; """)
+    openupgrade.logged_query(
+        cr,
+        """ \
+        UPDATE ir_sequence SET
+            prefix = REPLACE(prefix, '%%(year)s', '%%(range_year)s'),
+            suffix = REPLACE(suffix, '%%(year)s', '%%(range_year)s'),
+            use_date_range = TRUE
+        WHERE id IN (SELECT sequence_id FROM ir_sequence_date_range)
```
Comment (pedrobaeza) [inline addons/account/migrations/9.0.1.1/post-migration.py:360]: Use `...IN (SELECT DISTINCT(sequence_id) FROM...` for avoiding repetitive updates.

File: addons/account/migrations/9.0.1.1/post-migration.py (hunk 1 above)
Comment (StefanRijnhart) [reply addons/account/migrations/9.0.1.1/post-migration.py:344]: I *think* it looks more aligned in the logfile.

File: addons/account/migrations/9.0.1.1/post-migration.py (hunk 2 above)
Comment (StefanRijnhart) [reply addons/account/migrations/9.0.1.1/post-migration.py:360]: Possible SQL antipattern: https://www.quora.com/Should-I-use-DISTINCT-in-a-subquery-when-using-IN and this set of records is always very small so I'd rather leave this as is.

Comment (pedrobaeza) [reply addons/account/migrations/9.0.1.1/post-migration.py:360]: OK, got it. You're right

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**